Assets/UX/Scripts/LocalizationManager.cs

```diff
@@ -0,0 +1,260 @@
+using UnityEngine.ResourceManagement.AsyncOperations;
+
+public class LocalizationManager : MonoBehaviour
+{
+    public enum SupportedLanguages
+    {
+        ChineseSimplified,
+        English,
+        French,
+        German,
+        Hindi,
+        Italian,
+        Japanese,
+        Korean,
+        Portuguese,
+        Russian,
+        Spanish,
+        Tamil,
+        Telugu
+    }
+
+    public SupportedLanguages CurrentLocalizedLanguage;
+
+    const string k_ReasonTable = "Reasons";
+    const string k_UXTable = "UX";
+    const string k_InitializeKey = "INIT";
+    const string k_MotionKey = "MOTION";
+    const string k_LightKey = "LIGHT";
+    const string k_FeaturesKey = "FEATURES";
+    const string k_UnsupportedKey = "UNSUPPORTED";
+    const string k_NoneKey = "NONE";
+    const string k_MoveDeviceKey = "MOVE_DEVICE";
+    const string k_TapToPlaceKey = "TAP_TO_PLACE";
+    const string k_BodyKey = "FIND_BODY";
+    const string k_FaceKey = "FIND_FACE";
+    const string k_ImageKey = "FIND_IMAGE";
+    const string k_ObjectKey = "FIND_OBJECT";
+
+    public string localizedInit;
+    public string localizedMotion;
+    public string localizedLight;
+    public string localizedFeatures;
+    public string localizedUnsupported;
+    public string localizedNone;
+    public string localizedMoveDevice;
+    public string localizedTapToPlace;
+    public string localizedBody;
+    public string localizedFace;
+    public string localizedImage;
+    public string localizedObject;
+
+    bool m_LocalizationComplete = false;
+
+    public bool localizationComplete => m_LocalizationComplete;
+
+    [SerializeField]
+    TMP_FontAsset m_SimplifiedChineseFont;
+
+    public TMP_FontAsset simplifiedChineseFont
+    {
+        get => m_SimplifiedChineseFont;
+        set => m_SimplifiedChineseFont = value;
+    }
+
+    [SerializeField]
+    TMP_FontAsset m_JapaneseFont;
+
+    public TMP_FontAsset japaneseFont
+    {
+        get => m_JapaneseFont;
+        set => m_JapaneseFont = value;
+    }
+
+    [SerializeField]
+    TMP_FontAsset m_KoreanFont;
+    
+    public TMP_FontAsset koreanFont
+    {
+        get => m_KoreanFont;
+        set => m_KoreanFont = value;
+    }
+
+    [SerializeField]
+    TMP_FontAsset m_TamilFont;
+
+    public TMP_FontAsset tamilFont
+    {
+        get => m_TamilFont;
+        set => m_TamilFont = value;
+    }
+
+    [SerializeField]
+    TMP_FontAsset m_HindiFont;
+    
+    public TMP_FontAsset hindiFont
+    {
+        get => m_HindiFont;
+        set => m_HindiFont = value;
+    }
+
+    [SerializeField]
+    TMP_Text m_InstructionText;
+
+    public TMP_Text instructionText
+    {
+        get => m_InstructionText;
+        set => m_InstructionText = value;
+    }
+
+    [SerializeField]
+    TMP_Text m_ReasonText;
+
+    public TMP_Text reasonText
+    {
+        get => m_ReasonText;
+        set => m_ReasonText = value;
+    }
+
+    const int k_MaxAutoSizeSC = 70;
+
+    IEnumerator Start()

```

Comment:
You mentioned that the Localization package uses Addressables. In that case, I recommend using just the Addressables.LoadAssetAsync and then subscribe to the OnComplete event. If all the strings for a language are in the same group, these will be assigned once they are all loaded, so should be safe.